code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,826 @@
+	var/spider_role_summary = "UNDEFINED"
+	var/spider_placed = 0
+
+	// AI variables designed for use in procs
+	var/atom/cocoon_target // for queen and nurse
+	var/obj/machinery/atmospherics/unary/vent_pump/entry_vent // nearby vent they are going to try to get to, and enter
+	var/obj/machinery/atmospherics/unary/vent_pump/exit_vent // remote vent they intend to come out of
+	var/obj/machinery/atmospherics/unary/vent_pump/nest_vent // home vent, usually used by queens
+	var/fed = 0
+	var/travelling_in_vent = 0
+	var/list/enemies = list()
+	var/list/nibbled = list()
+	var/path_to_vent = 0
+	var/killcount = 0
+	var/busy = 0 // leave this alone!
+	var/spider_tier = 1 // 1 for red,gray,green. 2 for purple,black,white, 3 for prince, mother. 4 for queen, 5 for empress.
+	var/hasdroppedloot = 0
+	var/list/spider_special_drops = list()
+	var/attackstep = 0
+	var/attackcycles = 0
+	var/spider_myqueen = null
+	var/mylocation = null
+	var/chasecycles = 0
+
+	// Breathing, Pressure & Fire
+	// - No breathing / cannot be suffocated (spiders can hold their breath, look it up)
+	// - No pressure damage either - they have effectively exoskeletons
+	// - HOWEVER they can be burned to death!
+	// - Normal SPACE spiders should probably be immune to SPACE too, but meh, we try to leave the base spiders alone.
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 1500
+	heat_damage_per_tick = 5 //amount of damage applied if animal's body temperature is higher than maxbodytemp
+
+	// Xenobio Interactions
+	sentience_type = SENTIENCE_OTHER // prevents people from using a sentience potion on a TS to tame it
+	//gold_core_spawnable = CHEM_MOB_SPAWN_HOSTILE // this doesn't prevent a version with the "neutral" faction from gold slime extract + blood.
+	gold_core_spawnable = CHEM_MOB_SPAWN_INVALID // prevents this mob from being spawned by any xenobio or chem reactions.
+
+	// DEBUG OPTIONS & COMMANDS
+	var/spider_growinstantly = 0 // DEBUG OPTION, DO NOT ENABLE THIS ON LIVE. IT IS USED TO TEST NEST GROWTH/SETUP AI.
+	var/spider_debug = 0
+
+
+// --------------------------------------------------------------------------------
+// --------------------- TERROR SPIDERS: SHARED TARGETING & ATTACK CODE -----------
+// --------------------------------------------------------------------------------
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/ListTargets()
+	var/list/targets1 = list()
+	var/list/targets2 = list()
+	var/list/targets3 = list()
+	if (ai_type == 0)
+		// default, BE AGGRESSIVE
+		//var/list/Mobs = hearers(vision_range, src) - src // this is how ListTargets for /mob/living/simple_animal/hostile/ does it, but it is wrong, it ignores NPCs.
+		for(var/mob/living/H in view(src, vision_range))
+		//for(var/mob/H in Mobs)
+			if (H.stat == 2)
+				// dead mobs are ALWAYS ignored.
```

Comment:
You have these in a lot of places. If you want it to do nothing in an `if`, don't just leave it empty with a comment in there. Use early `return`s, or in this case `continue`.